compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -749,7 +749,7 @@ fn fn_sig_suggestion<'tcx>(
             Some(match ty.kind() {
                 ty::Param(_) if assoc.fn_has_self_parameter && i == 0 => "self".to_string(),
                 ty::Ref(reg, ref_ty, mutability) if i == 0 => {
-                    let reg = match &format!("{}", reg)[..] {
```

Comment:
this is weird... can we do something to simplify the logic?

Reply:
like this?

```rust
&*reg.to_string()
```

Reply:
no, sorry -- i meant to point out the body of the match block. it can avoid calling to_string twice.

Reply:
I improved it to avoid calling `to_string` twice.